tests/platform_tests/fwutil/conftest.py

```diff
@@ -0,0 +1,91 @@
+import tarfile
+import json
+import pytest
+
+from random import randrange
+
+from fwutil_common import show_firmware
+
+DUT_HOME="/home/admin"
+DEVICES_PATH="/usr/share/sonic/device"
+FS_PATH_TEMPLATE = "/host/image-{}/fs.squashfs"
+FS_RW_TEMPLATE = "/host/image-{}/rw"
+FS_WORK_TEMPLATE = "/host/image-{}/work"
+FS_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-fs"
+OVERLAY_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-overlay"
+
+def pytest_generate_tests(metafunc):
+    val = metafunc.config.getoption('--fw-pkg')
+    if 'fw_pkg_name' in metafunc.fixturenames and val is not None:
+        metafunc.parametrize('fw_pkg_name', [val], scope="module")
+
+@pytest.fixture(scope='module')
+def fw_pkg(fw_pkg_name):
+    with tarfile.open(fw_pkg_name, "r:gz") as f:
+        f.extractall()
+        with open('firmware.json', 'r') as fw:
+            fw_data = json.load(fw)
+            yield fw_data
+        for m in f.getmembers():
+            subprocess.call("rm -rf {}".format(m.name), shell=True)
+
+@pytest.fixture(scope='function')
+def random_component(duthost, fw_pkg):
+    chass = show_firmware(duthost)["chassis"].keys()[0]
+    components = fw_pkg["chassis"].get(chass, {})["component"].keys()
+
+    if len(components) == 0:
+        pytest.skip("No suitable components found in config file for platform {}.".format(duthost.facts['platform']))
+
+    return components[randrange(len(components))] 
+
+@pytest.fixture(scope='function')
+def host_firmware(localhost, duthost):
+    comm = "python3 -m http.server --directory {}".format(os.path.join(DEVICES_PATH, 
+        duthost.facts['platform']))
+    task, res = duthost.command(comm, module_ignore_errors=True, module_async=True)
+    yield "http://localhost:8000/"
```

Comment:
@alexrallen how about to also test `https`?

Reply:
Do we expect that this will test a part of the fwutil mechanism that isn't tested here? 

i.e. would we just be testing the python requests library here or actual functional code in fwutil that doesn't run unless its https?

Reply:
@alexrallen the same we can say about `http` - it's not a part of fwutil logic.
In any case it's just a suggestion - the choice is yours

Reply:
I see, I will probably hold off on that now, the current test exists primarily because it existed in the current fwutil test cases and is also intended just to enforce that however that path is being parsed, if it is a URL it gets parsed correctly more so than any networking. At least that is my understanding.